File: 134_Gas_Station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total, subsum, start = 0, float('inf'), 0
        for i in range(len(gas)):
            total += gas[i] - cost[i]
            if total < subsum:
                start = i + 1
                subsum = total
        return -1 if total < 0 else start % len(gas)

File: 134_Gas_Station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int total = 0, subsum = INT_MAX, start = 0;
        for (int i = 0; i < gas.size(); ++i) {
            total += gas[i] - cost[i];
            if (total < subsum) {
                start = i + 1;
                subsum = total;
            }
            cout << total << ", " << subsum << endl;
        }
        return total < 0 ? -1 : start % gas.size();
    }
};
